File: src/api/BoardingTracker.Domain/Entities/Vacancy.cs
namespace BoardingTracker.Domain.Entities
{
    public partial class Vacancy
    {
        public Vacancy()
        {
            Interviews = new HashSet<Interview>();
        }

        public int Id { get; set; }
        public string Title { get; set; } = null!;
        public string Description { get; set; } = null!;
        public decimal? Salary { get; set; }
        public int? SeniorityLevelId { get; set; }
        public int? VacancyTypeId { get; set; }
        public int? VacancyStatusId { get; set; }
        public int? RecruiterId { get; set; }

        public virtual Recruiter? Recruiter { get; set; }
        public virtual SeniorityLevel? SeniorityLevel { get; set; }
        public virtual VacancyStatus? VacancyStatus { get; set; }
        public virtual VacancyType? VacancyType { get; set; }
        public virtual ICollection<Interview> Interviews { get; set; }
    }
}

File: src/frontend/src/app/vacancies/models/vacancy.ts
import { SeniorityLevel } from "../../shared/models/seniority-level";
import { VacancyType } from "./vacancy-type";
import { VacancyStatus } from "./vacancy-status";
import { Recruiter } from "../../recruiters/models/recruiter";

export interface Vacancy {
  id: number;
  title: string;
  description: string;
  salary: number;
  seniorityLevel: SeniorityLevel;
  vacancyType: VacancyType;
  vacancyStatus: VacancyStatus;
  recruiter: Recruiter;
}
